stack-graphs/src/cycles.rs

```diff
@@ -111,10 +111,22 @@ where
         let key = path.key();
 
         let possibly_similar_paths = self.paths.entry(key).or_default();
-        for other_path in possibly_similar_paths.iter() {
-            match cmp(arena, path, other_path) {
-                Some(ord) if ord != Ordering::Less => return true,
-                _ => continue,
+        let mut idx = 0;
+        while idx < possibly_similar_paths.len() {
+            match cmp(arena, path, &possibly_similar_paths[idx]) {
+                Some(Ordering::Less) => {
+                    // the new path is betetr, remove the old one
+                    possibly_similar_paths.remove(idx);
+                    // keep `idx` which now points to the next element
```

Comment:
I don't love mutating a collection while iterating through it, but if we're going to do so, this seems like the reasonable way to do so. That said, it could probably use a comment above the loop noting what's going on here (and why).

Reply:
Good suggestion! In 07ca85d6bd593c26a03f34d577618e1b5d342337.